src/otx/core/model/module/action_classification.py

```diff
@@ -49,6 +49,10 @@ def on_test_epoch_end(self) -> None:
 
     def _log_metrics(self, meter: Accuracy, key: str) -> None:
         results = meter.compute()
+        if results is None:
+            msg = f"{meter} has no data to compute metric or there is an error computing metric"
+            raise RuntimeError(msg)
```

Comment:
If this error raising is repeating over all tasks can we wrap it up into a common util function?

Reply:
I am not sure whether it worth or not for 3 lines of code? We don't have proper file to put this common small func
![image](https://github.com/openvinotoolkit/training_extensions/assets/68103869/1e28784e-2e08-42bc-837c-2d0da51e16f0)

Reply:
How about adding this logic as a function of the `OTXLitModule` ? (i.e. `OTXLitModule.has_valid_output()` or `.has_nan_metric()`) Then, we can easily expand for all tasks.

Reply:
In addition, I investigated all the return types of metric functions:

1. `Accuracy`
![image](https://github.com/openvinotoolkit/training_extensions/assets/26541465/7b242606-d24c-4290-9bf6-5ae3d3672033)

2. `MeanAveragePrecision`
![image](https://github.com/openvinotoolkit/training_extensions/assets/26541465/178149cf-4ec8-4c84-8a1e-62c4eed7ec81)

3. `JaccardIndex`
![image](https://github.com/openvinotoolkit/training_extensions/assets/26541465/02c86c05-45d7-4ac6-98a7-c86d6ddd7c8f)

All of them is not `None`able. Are those what we really want?

Reply:
If you don't have any data in metric, then .compute() return None
I just wanted to add raising an error for this case. However, I don't want  to create separate method or function in utils for this. I guess, we can check metric in integration tests rather than raising some errors to exclude such problems in the future

Reply:
I agree with @sungmanc. ideally `_log_metric` would be part of the model interface. i see that each task has separate metric type. not sure if this is a good idea, though. 

Users might just want to try many metrics for tasks. For example, I might just want to try f1 score for a classification task, rather than accuracy. To be on the safe side, we could annotate the metric as `Metric`, which would be the parent class anyway?

Reply:
> If you don't have any data in metric, then .compute() return None I just wanted to add raising an error for this case. However, I don't want to create separate method or function in utils for this. I guess, we can check metric in integration tests rather than raising some errors to exclude such problems in the future

I still believe that there is a chance of refactoring to simplify this. I'm really against of adding code lines indefinitely without carefully. If you don't want to do it in this PR. I'll do.

Reply:
I am not against refactoring this. I'm just thinking how to do it better. Creating separate method in LitModule with 2 lines of code to check metric is None looks strange for me

Reply:
there will be multiple round of refactoring stages at the end anyway. That's what I was mentioning this morning